### PR TITLE
rbd-nbd: disallow mapping images >2TB in size

### PR DESCRIPTION
**Pull request by trociny**
Fixes: http://tracker.ceph.com/issues/17219
Signed-off-by: Mykola Golub <mgolub@mirantis.com>

### CODE REVIEW

File: qa/workunits/rbd/rbd-nbd.sh [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 POOL=rbd
 IMAGE=testrbdnbd$$
+TOOLARGE_IMAGE=${IMAGE}_large
```
Comment (dillaman) [inline qa/workunits/rbd/rbd-nbd.sh]: Minor: suggest adding an underscore between "TOO" and "LARGE" to make it clearer

**Comment by trociny** [comment]
@dillaman Thanks, updated.

File: src/tools/rbd_nbd/rbd-nbd.cc [hunk 2]
```diff
@@ -594,6 +594,14 @@ static int do_map()
   }
 
   size = info.size;
+
+  if (size > (1UL << 32) * 512) {
+    r = -EINVAL;
```
Comment (vshankar) [inline src/tools/rbd_nbd/rbd-nbd.cc]: minor: would E2BIG be a more appropriate errcode?

Comment (trociny) [inline src/tools/rbd_nbd/rbd-nbd.cc]: Not sure. E2BIG is usually used when user input parameters are too big. This looks a little different. Still I don't have a strong opinion. @dillaman what do you think?

Comment (dillaman) [inline src/tools/rbd_nbd/rbd-nbd.cc]: I don't think ```E2BIG``` is a correct error code for the reason Mykola stated. You might be able to get away with ```EFBIG``` for "file too large"

**Comment by trociny** [comment]
EFBIG looks much better! updated

**Comment by edisno** [comment]
I am having problems doing discard/trim on newly created rbd volumes.  The krbd client in linux 4.8.0 does not support certain default rbd flags, and rbd-fuse seems to not support trim/discard.  rbd-nbd does, but it is limited to 2TB and my volumes are much larger.  Any idea if rbd-nbd might support >2TB (properly) soon?  I read somewhere that this problem might only affect 32-bit systems, so would I risk data corruption by commenting out the check on my (64-bit) system?

**Comment by trociny** [comment]
@edisno See http://tracker.ceph.com/issues/18335
So we have plans to allow >2TB for not affected kernels. I am going to work on this ticket soon.

**Pull request merged**